packages/web3/src/contract/dapp-tx-builder.test.ts

```diff
@@ -16,142 +16,169 @@ You should have received a copy of the GNU Lesser General Public License
+        contractAddress: mockContractAddress,
+        methodIndex: 1,
+        args: ['456u'],
+        tokens: [{ id: testTokenId, amount: 100n }]
       })
-    ).toThrow('Invalid method index')
 
-    const commonParams = { contractAddress: randomContractAddress(), methodIndex: 0, args: [] }
-    expect(() =>
-      builder.callContract({
-        ...commonParams,
-        tokens: [{ id: 'invalid id', amount: 1n }]
-      })
-    ).toThrow('Invalid token id')
+      const result = builder.getResult()
+      expect(result.signerAddress).toBe(mockCallerAddress)
+      expect(result.tokens || []).toHaveLength(1)
+    })
+
+    it('should accumulate token amounts correctly', () => {
+      const builder = new DappTransactionBuilder(mockCallerAddress)
 
-    expect(() =>
       builder.callContract({
-        ...commonParams,
-        tokens: [{ id: randomContractId().slice(0, 60), amount: 1n }]
+        contractAddress: mockContractAddress,
+        methodIndex: 0,
+        args: [],
+        tokens: [{ id: testTokenId, amount: 100n }]
       })
-    ).toThrow('Invalid token id')
 
-    expect(() =>
       builder.callContract({
-        ...commonParams,
-        tokens: [{ id: ALPH_TOKEN_ID, amount: -1n }]
-      })
-    ).toThrow('Invalid token amount')
-
-    const result0 = builder.callContract({ ...commonParams }).getResult()
-    expect(result0.attoAlphAmount).toEqual(undefined)
-    expect(result0.tokens).toEqual([])
-
-    const tokenId0 = randomContractId()
-    const tokenId1 = randomContractId()
-    const result1 = builder
-      .callContract({
-        ...commonParams,
-        attoAlphAmount: ONE_ALPH,
-        tokens: [
-          { id: ALPH_TOKEN_ID, amount: ONE_ALPH },
-          { id: tokenId0, amount: ONE_ALPH },
-          { id: tokenId1, amount: 0n }
-        ]
-      })
-      .getResult()
-    expect(result1.attoAlphAmount).toEqual(ONE_ALPH * 2n)
-    expect(result1.tokens).toEqual([{ id: tokenId0, amount: ONE_ALPH }])
-
-    const result2 = builder
-      .callContract({
-        ...commonParams,
-        attoAlphAmount: 0n,
-        tokens: [
-          { id: tokenId0, amount: 0n },
-          { id: tokenId1, amount: ONE_ALPH }
-        ]
-      })
-      .callContract({
-        ...commonParams,
-        attoAlphAmount: ONE_ALPH,
-        tokens: [
-          { id: tokenId0, amount: ONE_ALPH },
-          { id: tokenId1, amount: ONE_ALPH }
-        ]
+        contractAddress: mockContractAddress,
+        methodIndex: 0,
+        args: [],
+        tokens: [{ id: testTokenId, amount: 150n }]
       })
-      .getResult()
-    expect(result2.attoAlphAmount).toEqual(ONE_ALPH)
-    expect(result2.tokens).toEqual([
-      { id: tokenId1, amount: ONE_ALPH * 2n },
-      { id: tokenId0, amount: ONE_ALPH }
-    ])
+
+      const result = builder.getResult()
+      const tokens = result.tokens || []
+      expect(tokens).toHaveLength(1)
+      if (tokens.length > 0) {
+        expect(tokens[0].amount).toBe(250n)
+      }
```

Comment:
This is a bit ambiguous series of statements

Reply:
ok i will fix that